src/pathfinding_igvc/src/AStar.cpp

```diff
@@ -6,10 +6,224 @@
  */
 
 #include <AStar.h>
+using namespace std;
+
+AStar::AStar(nav_msgs::OccupancyGrid occupancy_grid, GridPoint start,
+        GridPoint goal) {
+    this->_num_cols = occupancy_grid.info.width;
+    this->_num_rows = occupancy_grid.info.height;
+    this->_goal = goal;
+    this->_start = start;
+    this->_grid = occupancy_grid.data;
+
+    this->_cell_details = new CellDetail*[this->_num_rows];
+    for (int i = 0; i < this->_num_rows; i++) {
+        this->_cell_details[i] = new CellDetail[this->_num_cols];
+    }
+
+    /*
+     Create an open list
+     */
+    this->_open_list = set<GridPointWithScore>();
+
+    // Create a closed list and initialise it to false which means
+    // that no cell has been included yet
+    // This closed list is implemented as a boolean 2D array
+    this->_closed_list = new bool*[this->_num_rows];
+    for (int i = 0; i < this->_num_rows; i++) {
+        this->_closed_list[i] = new bool[this->_num_cols];
+    }
+}
 
 std::stack<AStar::GridPoint> AStar::run(nav_msgs::OccupancyGrid occupancy_grid,
                                         GridPoint start,
                                         GridPoint goal) {
-    // TODO Part 2
+    return AStar(occupancy_grid, start, goal).aStarSearch();
+}
+
+std::stack<AStar::GridPoint> AStar::aStarSearch() {
```

Comment:
`aStarSearch` is a bit redundant (since we're calling it on the `AStar` class). I'd suggest `search` or `findPath`